src/wp-includes/fonts/font-face/class-wp-font-face.php

```diff
@@ -0,0 +1,439 @@
+		 *
+		 * @param array $defaults {
+		 *     An array of required font-face properties and defaults.
+		 *
+		 *     @type string $provider     The provider ID. Default 'local'.
+		 *     @type string $font-family  The font-family property. Default empty string.
+		 *     @type string $font-style   The font-style property. Default 'normal'.
+		 *     @type string $font-weight  The font-weight property. Default '400'.
+		 *     @type string $font-display The font-display property. Default 'fallback'.
+		 * }
+		 */
+		$this->font_face_property_defaults = apply_filters( 'wp_font_face_property_defaults', $this->font_face_property_defaults );
+
+		if (
+			function_exists( 'is_admin' ) && ! is_admin()
+			&&
+			function_exists( 'current_theme_supports' ) && ! current_theme_supports( 'html5', 'style' )
+		) {
+			$this->style_tag_attrs = array( 'type' => 'text/css' );
+		}
+	}
+
+	/**
+	 * Generates and prints the `@font-face` styles for the given fonts.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[][] $fonts {
+	 *     Optional. The font-families and their font variations. Default empty array.
+	 *
+	 *     @type string $font-family => array[] $variations {
+	 *         Optional. An associated array of font variations for this font-family.
+	 *         Each variation has the following structure.
+	 *
+	 *         @type array $font_variation {
+	 *             @type string          $font-family             The font-family property.
+	 *             @type string|string[] $src                     The URL(s) to each resource containing the font data.
+	 *             @type string          $font_style              Optional. The font-style property. Default 'normal'.
+	 *             @type string          $font-weight             Optional. The font-weight property. Default '400'.
+	 *             @type string          $font-display            Optional. The font-display property. Default 'fallback'.
+	 *             The following font-face properties are optional and will not be generated if not provided.
+	 *             @type string          $ascent-override         Optional. The ascent-override property.
+	 *             @type string          $descent-override        Optional. The descent-override property.
+	 *             @type string          $font-stretch            Optional. The font-stretch property.
+	 *             @type string          $font-variant            Optional. The font-variant property.
+	 *             @type string          $font-feature-settings   Optional. The font-feature-settings property.
+	 *             @type string          $font-variation-settings Optional. The font-variation-settings property.
+	 *             @type string          $line-gap-override       Optional. The line-gap-override property.
+	 *             @type string          $size-adjust             Optional. The size-adjust property.
+	 *             @type string          $unicode-range           Optional. The unicode-range property.
+	 *         }
+	 *     }
```

Comment:
See my other comment, I think for maintenance purposes and to prevent things getting out of sync, we should keep these docs in a single place, either here or in `wp_print_font_faces()`. The respective other function should then refer to that, for example like this:
```suggestion
	 * @param array[][] $fonts Optional. The font-families and their font variations.
	 *                         See {@see wp_print_font_faces()} for the supported fields.
	 *                         Default empty array.
```

Reply:
Thank you @felixarntz. Great suggestion! Committed your suggestion in commit https://github.com/WordPress/wordpress-develop/pull/5051/commits/f6c42668cf426456a5d5e36e1ad5d15636dc523e ✅